wodles/aws/aws_s3.py

```diff
@@ -845,22 +846,67 @@ def single_element_list_to_dictionary(my_event):
 
         return event
 
-    def decompress_file(self, log_key):
-        def decompress_gzip(raw_object):
-            # decompress gzip file in text mode.
-            try:
-                # Python 3
-                return gzip.open(filename=raw_object, mode='rt')
-            except TypeError:
-                # Python 2
-                return gzip.GzipFile(fileobj=raw_object, mode='r')
+    def _decompress_gzip(self, raw_object: io.BytesIO):
+        """
+        Method that decompress gzip compressed data.
+
+        Parameters
+        ----------
+        raw_object : io.BytesIO
+            Buffer with the gzip compressed object.
+
+        Returns
+        -------
+        file_object
+            Decompressed object.
+        """
+        try:
+            gzip_file = gzip.open(filename=raw_object, mode='rt')
+            # Ensure that the file is not corrupted by reading from it
+            gzip_file.read()
+            gzip_file.seek(0)
```

Comment:
```suggestion
```
This function after the `read` call will not raise any errors as it merely moves the position within the stream.

Reply:
I set it that way because if you don't, you are returning a file with the read pointer not set to the beginning of it, which will be troublesome when trying to read events from it.

Reply:
Tests did not fail after the removal of this line, however, we will undo the change just to be sure and we will keep the original implementation using `seek` instead.